pkg/geodata/geodata_test.go

```diff
@@ -446,3 +449,237 @@ func normSQL(sql string) string {
+			if !assert.Equal(t, test.wantIncludes, includes) {
+				return
+			}
+
+			// parse wantcols into a ValueSet so we can compare with newset
+			wantset, err := where.ParseMultiArgs(test.wantCols)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			// verify resulting ValueSet after extraction
+			assert.Equal(t, wantset, newset)
+		})
+	}
+}
+
+func TestValidateAllToken_Error(t *testing.T) {
+	// catch error when a ALL is given more than once or in a range
+	var tests = map[string]struct {
+		cols []string // input cols as received from query string
+	}{
+		"ALL is given more than once": {
+			[]string{"ALL", "ALL}"},
+		},
+		"ALL is low part of a range": {
+			[]string{"ALL...high"},
+		},
+		"ALL is high part of a range": {
+			[]string{"low...ALL"},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			// parse cols into a ValueSet
+			set, err := where.ParseMultiArgs(test.cols)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			// call validator
+			err = geodata.ValidateAllToken(set)
+			assert.Error(t, err)
+		})
+	}
+}
+
+func TestValidateAllToken_OK(t *testing.T) {
+	var tests = map[string]struct {
+		cols []string // input cols as received from query string
+	}{
+		"ALL is not given at all": {
+			[]string{"foo"},
+		},
+		"ALL is only token": {
+			[]string{"ALL"},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			// parse cols into a ValueSet
+			set, err := where.ParseMultiArgs(test.cols)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			// call validator
+			err = geodata.ValidateAllToken(set)
+			assert.NoError(t, err)
+		})
+	}
+}
+
+func TestFixgeotype_Error(t *testing.T) {
+	// catch invalid geotypes
+	_, err := geodata.FixGeotype("foo")
+	assert.Error(t, err)
+}
+
+func TestFixgeotype_OK(t *testing.T) {
+	// verify geotypes are fixed
+	geotype, err := geodata.FixGeotype("country")
+	assert.NoError(t, err)
+	assert.Equal(t, geotype, "Country")
+}
+
+func TestMapGeotypes_Error(t *testing.T) {
+	// catch error when a geotype isn't valid
+	var tests = map[string]struct {
+		geos []string // input geos as received from query string
+	}{
+		"invalid single geotype": {
+			[]string{"foo"},
+		},
+		"geotype as low part of range": {
+			[]string{"lsoa...foo"},
+		},
+		"geotype as high part of range": {
+			[]string{"foo...lsoa"},
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			// parse geos into a ValueSet
+			set, err := where.ParseMultiArgs(test.geos)
+			if !assert.NoError(t, err) {
+				return
+			}
+
+			// expect validation error
+			_, err = geodata.MapGeotypes(set)
+			assert.Error(t, err)
+		})
+	}
+}
+
+func TestMapGeotypes_OK(t *testing.T) {
+	var tests = map[string]struct {
+		geos []string // input geos as received from query string
+		want []string // expected fixed geos
+	}{
+		"single geotype": {
+			[]string{"lsoa"},
+			[]string{"LSOA"},
+		},
+		"multiple geotyupes": {
```

Comment:
typo, geotypes

Reply:
I claim to be a touch typist, and I claim to be able to read. Something doesn't add up.